client/src/components/Atoms/Steps.tsx

```diff
@@ -143,3 +149,17 @@ export const StepActions: React.FC<{
     {right || <div />}
   </StepActionsRow>
 )
+
+// Utility function to compute step state for the most common step list pattern:
+// - a single current step
+// - all previous steps are complete
+// - all subsequent steps are incomplete
```

Comment:
Nice! I like this compromise - leaving state management up to consumers but providing utilities for common cases. Similar in spirit to composable components